src/codeManager.ts

```diff
@@ -162,7 +167,15 @@ export class CodeManager {
         fs.writeFileSync(this._codeFile, content);
     }
 
-    private getExecutor(languageId: string, fileExtension: string): string {
+    private async getSubExecutor(executor: any): Promise<string> {
+        let subExecutorId = await vscode.window.showQuickPick(Object.keys(executor), { placeHolder: "Type or select command to run" });
+        if (subExecutorId !== undefined) {

```

Comment:
Need to handle the case when subExecutorId is `undefined` (user does select command to run)